yabt/builders/cpp.py

```diff
@@ -260,21 +272,37 @@ def compile_cc(build_context, compiler_config, buildenv, sources,
         if compiler_config.use_fdebug_prefix_map_flag:
             # Store relative paths (instead of absolute) in debugger symbols
             # when in debug mode (with gcc and clang, it is harmless otherwise)
-            special_flags.extend(['-fdebug-prefix-map=%s=.' %
-                                  buildenv_workspace])
-
-        compile_cmd = (
-            [compiler_config.compiler, '-o', obj_file, '-c'] +
-            compiler_config.compile_flags +
-            ['-I{}'.format(path) for path in include_paths] +
-            special_flags +
-            [join(buildenv_workspace, src)])
+            special_flags.extend(
+                ['-fdebug-prefix-map=%s=.' % buildenv_workspace])
+        if compiler_config.run_clang_tidy:
+            special_flags.append('-MJ%s' % buildenv_path_to_compile_commands)
+        compile_cmd = ([compiler_config.compiler, '-o', obj_file, '-c'] +
+                       compiler_config.compile_flags +
+                       ['-I{}'.format(path) for path in include_paths] +
+                       special_flags + [join(buildenv_workspace, src)])
         # TODO: capture and transform error messages from compiler so file
         # paths match host paths for smooth(er) editor / IDE integration
         build_context.run_in_buildenv(buildenv, compile_cmd, cmd_env)
+        if compiler_config.run_clang_tidy:
+            # aggregate compile commands
+            with open(host_path_to_compile_commands, 'r') as f:
+                curr_compile_commnads = json.loads(f.read().strip()[:-1])
+                compile_commands.append(curr_compile_commnads)
         objects.append(
             join(relpath(workspace_dir, build_context.conf.project_root),
                  obj_rel_path))
+    if compiler_config.run_clang_tidy:
+        # output aggregate json
+        with open(host_path_to_compile_commands, 'w') as f:
+            json.dump(compile_commands, f)
+        # run clang-tidy
+        for src in sources:
+            clang_tidy_cmd = ([
+                compiler_config.clang_tidy, '-config-file',
+                compiler_config.clang_tidy_config, '-p', buildenv_workspace,
+                join(buildenv_workspace, src)
+            ])
+            build_context.run_in_buildenv(buildenv, clang_tidy_cmd, cmd_env)
```

Comment:
I suggest adding a test to this functionality (maybe one with good cpp file and one with lint error)
I am not sure how easy it will be since I think our other cpp tests are with g++ and not clang. Let me know if it gets too complicated